### PR TITLE
IRIS-3830 | cleaned up tests and page objects, added new test cases for mobile view

### PR DESCRIPTION
**Pull request by tortila**
[Ticket](https://wikia-inc.atlassian.net/browse/IRIS-3830)

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 1]
```diff
@@ -1,13 +1,25 @@
 package com.wikia.webdriver.elements.mercury.components.discussions.desktop;
 
+import com.google.common.base.Predicate;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'com.google.common.base.Predicate'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
 import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
 
+import lombok.Getter;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'lombok.Getter'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+import org.openqa.selenium.By;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'org.openqa.selenium.By'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+import org.openqa.selenium.WebDriver;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'org.openqa.selenium.WebDriver'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 5]
```diff
@@ -1,13 +1,25 @@
...
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'org.openqa.selenium.support.ui.ExpectedConditions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 6]
```diff
@@ -1,13 +1,25 @@
...
+import org.openqa.selenium.support.ui.FluentWait;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'org.openqa.selenium.support.ui.FluentWait'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 7]
```diff
@@ -1,13 +1,25 @@
...
+import org.openqa.selenium.support.ui.WebDriverWait;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'org.openqa.selenium.support.ui.WebDriverWait'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java [hunk 8]
```diff
@@ -1,13 +1,25 @@
...
+
+import javax.annotation.Nullable;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/Promoting.java]: MINOR Remove this unused import 'javax.annotation.Nullable'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java [hunk 9]
```diff
@@ -88,8 +88,7 @@ private void deleteCategoriesMobile() {
   }
 
   private void deleteCategories(String wikiName) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java]: MAJOR Remove this unused private "deleteCategories" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 10]
```diff
@@ -41,8 +41,7 @@
    * @return post that was created
    */
   private PostEntity.Data setUp(String wikiName) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR Remove this unused private "setUp" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 11]
```diff
@@ -533,6 +495,34 @@ public String switchToNewBrowserTab() {
...
+  }
+
+  private String switchToNewTab(String parentTab) {
+    String newTab = getNewTab(parentTab);
+    driver.switchTo().window(newTab);
+    return newTab;
+  }
+
+  private void waitForLinkOpenedInNewTab(WebElement link) {
+    int tabs = driver.getWindowHandles().size();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Remove this unused "tabs" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 12]
```diff
@@ -18,14 +19,7 @@
...
-import org.openqa.selenium.Dimension;
-import org.openqa.selenium.JavascriptExecutor;
-import org.openqa.selenium.NoSuchElementException;
-import org.openqa.selenium.Point;
-import org.openqa.selenium.TimeoutException;
-import org.openqa.selenium.WebDriverException;
-import org.openqa.selenium.WebElement;
+import org.openqa.selenium.*;
 import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.FindBy;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MINOR Remove this unused import 'org.openqa.selenium.support.FindBy'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 28 issues
* MAJOR 13 major
* MINOR 15 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BasePageObject.java#L109](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L109): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [CategoriesTests.java#L119](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L119): Remove this unused private "setUp" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L454](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L454): Remove this unused private "openPageAndSelectCategoryOnMobile" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L461](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L461): Remove this unused private "openPageAndSelectCategoryOnDesktop" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L466](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L466): Remove this unused private "postsOnPageAreOnlyFromOneCategory" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L500](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L500): Remove this unused private "addCategoryRemotely" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L303](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L303): Remove this unused private "assertThatPostWasAddedWith" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L343](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L343): Remove this unused private "assertPostWithWeirdCharactersDisplayedOnPostsListPage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L357](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L357): Remove this unused private "assertPostWithWeirdCharactersDisplayedOnPostDetailsPage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [DeleteAllPostsByUserTests.java#L218](https://github.com/Wikia/selenium-tests/blob/2164005d7a6b88a590f7fc1b01f9084924294da2/src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java#L218): Remove this unused private "deleteAllOptionNotVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java (hunk 9 above)
Comment (tortila) [inline src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java]: invalid

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 11 above)
Comment (tortila) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: invalid

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues
* MAJOR 13 major
* MINOR 6 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BasePageObject.java#L108](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L108): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [CategoriesTests.java#L119](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L119): Remove this unused private "setUp" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L454](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L454): Remove this unused private "openPageAndSelectCategoryOnMobile" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L461](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L461): Remove this unused private "openPageAndSelectCategoryOnDesktop" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L466](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L466): Remove this unused private "postsOnPageAreOnlyFromOneCategory" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L500](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L500): Remove this unused private "addCategoryRemotely" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L303](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L303): Remove this unused private "assertThatPostWasAddedWith" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L343](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L343): Remove this unused private "assertPostWithWeirdCharactersDisplayedOnPostsListPage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L357](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L357): Remove this unused private "assertPostWithWeirdCharactersDisplayedOnPostDetailsPage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [DeleteAllPostsByUserTests.java#L218](https://github.com/Wikia/selenium-tests/blob/3e80b9ce217ae6ca440a1ec8a865092fcbef1811/src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java#L218): Remove this unused private "deleteAllOptionNotVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java (hunk 10 above)
Comment (tortila) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: invalid

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 11 above)
Comment (slayful) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: `tabs` -> `initialTabNumber`?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 13]
```diff
@@ -533,6 +494,34 @@ public String switchToNewBrowserTab() {
...
+    String newTab = getNewTab(parentTab);
+    driver.switchTo().window(newTab);
+    return newTab;
+  }
+
+  private void waitForLinkOpenedInNewTab(WebElement link) {
+    int tabs = driver.getWindowHandles().size();
+    link.click();
+    new WebDriverWait(driver, TIMEOUT_PAGE_REGISTRATION)
+      .until((Predicate<WebDriver>) input -> driver.getWindowHandles().size() > tabs);
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: Extract the `driver.getWindowHandles().size()` into another method `getTabNumber`?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 14]
```diff
@@ -533,6 +494,34 @@ public String switchToNewBrowserTab() {
...
+  }
+
+  protected void openLinkInNewTab(WebElement link) {
+    String oldTab = driver.getWindowHandle();
+    waitForLinkOpenedInNewTab(link);
+    switchToNewTab(oldTab);
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: `oldTab` -> `currentTab`?

File: src/test/java/com/wikia/webdriver/testcases/discussions/PromotingTests.java [hunk 15]
```diff
@@ -1,76 +1,81 @@
...
 
-  private void discussionsAppPromotionUnitPresentOnPage() {
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void anonUserOnMobileCanSeeAppPromotion() {
     Promoting promoting = findPromoting();
-    Assertion.assertTrue(promoting.isAppleLinkDisplayed());
-    Assertion.assertTrue(promoting.isGooglePlayLinkDisplayed());
-    Assertion.assertEquals(promoting.isPromotionAppTextDisplayed(), "Take your fandom with you, download the app today!");
+    Assertion.assertTrue(promoting.isMobileBannerDisplayed());
+    Assertion.assertStringContains(promoting.getPromotionAppMobileText(), MOBILE_PROMOTION_TEXT);
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/testcases/discussions/PromotingTests.java]: Perhaps you can use static imports for everything form Assertion?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 16]
```diff
@@ -533,6 +494,38 @@ public String switchToNewBrowserTab() {
     return driver.getCurrentUrl();
   }
 
+  private int getTabsCount() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 17]
```diff
@@ -533,6 +494,38 @@ public String switchToNewBrowserTab() {
...
+  }
+
+  private String switchToNewTab(String parentTab) {
+    String newTab = getNewTab(parentTab);
+    driver.switchTo().window(newTab);
+    return newTab;
+  }
+
+  private void waitForLinkOpenedInNewTab(WebElement link) {
+    int initialTabsNumber = driver.getWindowHandles().size();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* MAJOR 14 major
* MINOR 6 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BasePageObject.java#L108](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L108): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. MAJOR [CategoriesTests.java#L119](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L119): Remove this unused private "setUp" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L454](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L454): Remove this unused private "openPageAndSelectCategoryOnMobile" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L461](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L461): Remove this unused private "openPageAndSelectCategoryOnDesktop" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L466](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L466): Remove this unused private "postsOnPageAreOnlyFromOneCategory" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CategoriesTests.java#L500](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CategoriesTests.java#L500): Remove this unused private "addCategoryRemotely" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L303](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L303): Remove this unused private "assertThatPostWasAddedWith" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L343](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L343): Remove this unused private "assertPostWithWeirdCharactersDisplayedOnPostsListPage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [CreatingPostTests.java#L357](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/CreatingPostTests.java#L357): Remove this unused private "assertPostWithWeirdCharactersDisplayedOnPostDetailsPage" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MAJOR [DeleteAllPostsByUserTests.java#L218](https://github.com/Wikia/selenium-tests/blob/e7509669753a05456a3573f55623669e41bf557a/src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java#L218): Remove this unused private "deleteAllOptionNotVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 17 above)
Comment (tortila) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: invalid

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 16 above)
Comment (tortila) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: invalid

**Pull request merged**